File: java/src/main/java/mx/iteso/adapter/WordDoc.java
package mx.iteso.adapter;

/**
 * A word document.
 */
public class WordDoc implements IWordDocument {
    /**
     * The format of the document.
     */
    private Format format;

    /**
     * The image of the document.
     */
    private Image img;

    /**
     * The license of microsoft.
     */
    private MSLicense license;

    /**
     * The version of the document.
     */
    private double version;

    /**
     * The recent version of the document.
     */
    private static final double RECENT_VERSION = 1.1;

    /**
     * The valid license number.
     */
    private static final int LICENSE_VALID_NUMBER = 10000;


    /**
     * Creator.
     * @param lscNumber the license number.
     */
    public WordDoc(final int lscNumber) {
        this.format = new Format();
        this.img = new Image("img.png");
        this.license = new MSLicense(lscNumber);
        this.version = RECENT_VERSION;
    }

    @Override
    public final Format getFormat() {
        return this.format;
    }

    @Override
    public final Image getBackground() {
        return this.img;
    }

    @Override
    public final void setMSOfficeVersion(final float msOfficeVersion) {
        this.version = msOfficeVersion;
    }

    @Override
    public final MSLicense getLicense() {
        return this.license;
    }

    @Override
    public final boolean restrictEditIfLicenseIsInvalid(final MSLicense msLicense) {
        if (msLicense.getNumber() > LICENSE_VALID_NUMBER) {
            return false;
        }
        return true;
    }
}

File: python/src/main/python/WordDoc.py
from IWordDocument import *
from Format import *
from Image import *
from MSLicense import *

class WordDoc(IWordDocument):

    def __init__(self, lsc_number):
        self.format = Format()
        self.img = Image("img.png")
        self.license = MSLicense(lsc_number)
        self.version = 1.1
    
     
    def getFormat(self):
        return self.format

    
    def getBackground(self):
        return self.img

    
    def setMSOfficeVersion(self, ms_office_version):
        self.version = ms_office_version

    
    def getLicense(self):
        return self.license

    
    def restrictEditIfLicenseIsInvalid(self, ms_license):
        if (ms_license.getNumber() > 10000):
            return False
        return True